x-pack/metricbeat/module/gcp/metrics/metricset.go

```diff
@@ -198,16 +198,16 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) (err erro
 func (m *MetricSet) eventMapping(ctx context.Context, tss []timeSeriesWithAligner, sdc metricsConfig) ([]mb.Event, error) {
 	e := newIncomingFieldExtractor(m.Logger(), sdc)
 
-	var gcpService = gcp.NewStackdriverMetadataServiceForTimeSeries(nil)
+	var metadataService gcp.MetadataService
```

Comment:
You can change it to this to avoid later renamings.
```suggestion
	var gcpService gcp.MetadataService
```